sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-processing.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Injectable } from "@nestjs/common";
+import {
+  DisbursementReceipt,
+  DisbursementReceiptValue,
+  DisbursementSchedule,
+  User,
+} from "../../database/entities";
+import { InjectLogger } from "../../common";
+import { LoggerService } from "../../logger/logger.service";
+import {
+  ConfigService,
+  DisbursementScheduleService,
+  DisbursementReceiptService,
+} from "../../services";
+import { ESDCIntegrationConfig } from "../../types";
+import { DisbursementReceiptIntegrationService } from "./disbursement-receipt-integration.service";
+import { ProcessSFTPResponseResult } from "../models/esdc-integration.model";
+import { DisbursementReceiptDownloadResponse } from "./models/disbursement-receipt-integration.model";
+
+/**
+ * Manages the process to import disbursement receipt files which are
+ * downloaded from SFTP location.
+ */
+@Injectable()
+export class DisbursementReceiptProcessingService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly integrationService: DisbursementReceiptIntegrationService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+    private readonly disbursementReceiptService: DisbursementReceiptService,
+  ) {
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+
+  /**
+   * Process all the available disbursement receipt files in SFTP location.
+   * Once the file is processed, it gets deleted.
+   * @param auditUserId
+   * @returns Summary details of the processing.
+   */
+  async process(auditUserId: number): Promise<ProcessSFTPResponseResult> {
```

Comment:
I would say that usually, this method would return one `ProcessSFTPResponseResult` for each processed file. Does keeping all the logs about all the files in one single log makes it cleaner?